### PR TITLE
added signup login functionality

### PR DESCRIPTION
**Pull request by aishwarya-bm**
signup, login functionality added.
added option to show menu on a video to like/add to playlist/watch  later

https://user-images.githubusercontent.com/66695068/161448761-5cbd144c-5bc5-461e-9143-fe9e8a777cd1.mp4

### CODE REVIEW

File: src/pages/videolist/Videolist.jsx [hunk 1]
```diff
@@ -41,31 +44,55 @@ export function Videolist() {
...
+                      <img src={thumnailHigh.url} alt="card-img" />
                     </div>
+                  </Link>
 
-                    <button className="btn btn-link icon-more">
+                  <div className="card-head d-flex p-rel">
+                    <Link to={`/explore/video/${_id}`}>
+                      <div className="card-header children-stacked">
+                        <div className="card-title">
+                          {"title title title title title title title title"}
```
Comment (srejitk) [inline src/pages/videolist/Videolist.jsx:57]: I'm assuming you're going to get the title here, replace it with the video title here

**Review by srejitk — APPROVED** [review]
Good Work. Really liked how you handled the various server responses, and input validations.

File: src/contexts/login-context/login-utils.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        navigate("/");
+        Toast({
+        message: "Signed in successfully.",
+        type: "success",
+      });
+      } else {
+        Toast({
+        message: "Invalid credentials. Please try again.",
+        type: "error",
+      });
```
Comment (sadathshariff) [inline src/contexts/login-context/login-utils.js:106]: The Catch would handle this anyway if some error occurs

**Review by sadathshariff — APPROVED** [review]
Looks Good, like how you are handling the form validations

File: src/pages/videolist/Videolist.jsx (hunk 1 above)
Comment (aishwarya-bm) [reply src/pages/videolist/Videolist.jsx:57]: Hey yes this is until i implement the features, will change in upcoming pushes

File: src/contexts/login-context/login-utils.js (hunk 2 above)
Comment (aishwarya-bm) [reply src/contexts/login-context/login-utils.js:106]: Sometime the status in try could be other than 200 so to handle that I have added this.

**Pull request merged**